Source/Core/Common/FileUtil.cpp

```diff
@@ -276,33 +279,68 @@ bool DeleteDir(const std::string& filename)
   return false;
 }
 
+// Repeatedly invokes func until it returns true or max_attempts failures.
+// Waits after each failure, with each delay doubling in length.
+template <typename FuncType>
+static bool AttemptMaxTimesWithExponentialDelay(const FuncType& func, int max_attempts,
+                                                std::chrono::milliseconds delay)
+{
+  for (int failed_attempts = 0; failed_attempts < max_attempts; ++failed_attempts)
+  {
+    if (func())
+    {
+      return true;
+    }
+    if (failed_attempts + 1 < max_attempts)
+    {
+      INFO_LOG_FMT(COMMON, "Rename attempt failed, delaying for {} milliseconds", delay.count());
```

Comment:
Since this is a generic utility function, I think the name of the operation should be passed as a parameter to this function:

```suggestion
      INFO_LOG_FMT(COMMON, "{} attempt failed, delaying for {} milliseconds", operation_name, delay.count());
```